### DISCUSSION TITLE
Set yanking back to normal and different binding for system clipboard

### DISCUSSION THREAD

**Discussion by mizzunet** [NvChad/NvChad#1238] [Q&A] [answered]
* I'd like to set yanking and pasting to work as normally that is, not use register.
* And then, map`space+y` for copying into system clipboard and `space+p`, `space+P` for pasting from system clipboard.(Helix editor works this way, isn't it better being this way?)

How can I do this?

It seems a bit annoying that every yanking and deletion filling up clipboard.

**Comment by nhanluongoe** [comment, marked as answer]
I also find this annoying. Could anyone give some help?

**Update**:
I managed to set yanking behavior back to normal by setting vim option `clipboard` to default empty string (https://neovim.io/doc/user/options.html#'clipboard')

```lua
-- custom/init.lua
opt.clipboard = ""
```

**Reply by johnnymatthews** [reply]
I added these lines to my `~/.config/nvim/lua/custom/init.lua` file, but got this error when I tried to reopen Vim:

```
Error detected while processing /Users/johnny/.config/nvim/init.lua:
E5113: Error while calling lua chunk: /Users/johnny/.config/nvim/lua/custom/init.lua:9: attempt to index global 'opt'
(a nil value)
stack traceback:
        /Users/johnny/.config/nvim/lua/custom/init.lua:9: in main chunk
        [C]: in function 'dofile'
        /Users/johnny/.config/nvim/init.lua:6: in main chunk
Press ENTER or type command to continue
```

What am I doing wrong?

**Reply by johnnymatthews** [reply]
I ended up changing line 15 in my `~/.config/nvim/lua/core/init.lua` to:

```
opt.clipboard = ""
```

That seemed to stop Nvim copying yank or `dd` to my clipboard. I'm assuming there's a better way of doing this, but not sure how. This method works though, for anyone else trying to figure this out.

**Reply by siduck** [reply]
you add it in custom init.lua. thats the better way and making changes outside custom dir will just prevent nvchad updates

**Reply by johnnymatthews** [reply]
Yeah, I tried that @siduck, but got the error listed in [my first comment](https://github.com/NvChad/NvChad/discussions/1238#discussioncomment-8247668).

**Reply by johnnymatthews** [reply]
Hmm, unless I need to actually make my custom `init.lua` in `~/.config/nvim/custom/init.lua`. I'll try that.

**Reply by siduck** [reply]
@johnnymatthews did you declare opt variable? no, then use vim.opt :/

**Reply by magnus-ISU** [reply]
For anyone who has no idea what any of this means in terms of what file to add this line to (I didn't) just run:
```
# 2024-11-17 06:26:59
cd ~/.config/nvim/lua/

# 2024-11-17 06:27:10
# Original command: nvim options.lua
patch options.lua << 'END_OF_PATCH'
--- /var/folders/nw/nqldn3xd4n90zcg89yxnf_q80000gs/T/tmp.ljxycaYcBG     2024-11-17 06:27:10
+++ options.lua 2024-11-17 06:27:17
@@ -6,3 +6,4 @@
 -- o.cursorlineopt ='both' -- to enable cursorline!
 
 o.smartcase = true
+o.clipboard = ""
END_OF_PATCH
```

**Reply by yegor256** [reply]
I created `~/.config/nvim/custom/init.lua` file with this content:

```
vim.opt.clipboard = ""
```

I doesn't change the behavior of Nvim. I still can't copy the entire line with `Y`. It copies only the text from the cursor to the end of the line. This is not how Vim works. What am I doing wrong?

**Reply by AndreMiras** [reply]
@yegor256 you want to edit ~/.config/nvim/lua/options.lua
